Прикрой меня тестом

VAVPSD opened the pull request:
---
:mortar_board: [Прикрой меня тестом](https://up.htmlacademy.ru/react-individual/3/user/2460349/tasks/16)

VAVPSD committed:
refactor:переключение избранного вынесено в хук, логика сортировки по городу на странице избранных  вынесена в селектор, исправлен нейминг

VAVPSD committed:
refactor: исправление нейминга в api-actions

VAVPSD committed:
feat: отображаются последние комментарии в порядке от новых к старым, ограниченные по лимиту

VAVPSD committed:
feat: блокировка всей формы при отправке комментария

VAVPSD committed:
feat: перенаправление на главную с логина для авторизованных пользователей

VAVPSD committed:
refactor: вынесены типы, перемещён роутер, приведён код к единому стилю

VAVPSD committed:
test: добавлены тесты для селекторов и слайсов

VAVPSD committed:
test: добавлены тесты для api-actions, вынесены данные для тестов в отдельный фаил

VAVPSD committed:
test: добавлены тесты для простых компонентов

VAVPSD committed:
test: добавлены тесты маршрутизации(без оффера)

potyoma commented on src/hooks/use-favorites-toggle.ts:
```diff
@@ -0,0 +1,36 @@
+  const isAuth = authStatus === AuthorizationStatus.Auth;
+
+  const handleToggle = async (data: FavoritesData) => {
+    if (!isAuth && context !== 'favorites') {
```
nit: лучше бы перенести это в thunk

potyoma commented on src/hooks/use-favorites-toggle.ts:
```diff
@@ -0,0 +1,36 @@
+        break;
+    }
+
+    dispatch(fetchOffers());
```
эта ревалидация должна происходить только на главной

potyoma commented on src/pages/offer-screen/components/reviews-form.tsx:
```diff
@@ -53,26 +56,28 @@ function ReviewsForm() {
+          <p className="reviews__help">
+            To submit review please make sure to set <span className="reviews__star">rating</span> and describe your stay with at least <b className="reviews__text-amount">50 characters</b>.
+          </p>
+          <button className="reviews__submit form__submit button" type="submit" disabled={ review.comment.length < 50 || review.comment.length > 300 || review.rating === 0}>
```
лучше это вычисление вынести из разметки

potyoma commented on src/store/reviews-slice/reviews-selectors.test.ts:
```diff
@@ -0,0 +1,41 @@
+import { NameSpace } from '../../const';
+import { mockComments } from '../../mock/test-data';
+
+const MAX_VISIBLE_COMMENTS = 10;
```
константы можно переиспользовать

VAVPSD committed:
fix: улучшена логика получения и обновления данных

- обновление предложений перенесено на main
- обновлён селектор подсчёта избранных
- добавлена ссылка на оффер в листе избранного
- логика дизейбла формы вынесена из компонента
- dispatch fetchOffers добавлен в toggleFavoriteStatus

VAVPSD committed:
fix: добавлены пропущенные требования из ТЗ

VAVPSD committed:
test: удалена часть тестов
